doc/nrf/test_and_optimize.rst

```diff
@@ -1,17 +1,165 @@
+.. _gs_testing:
+.. _serial_terminal_connect:
+.. _testing_rtt:
+.. _testing_rtt_connect:
+.. _putty:
+.. _testing_vscode:
 
 Testing and optimization
 ########################
 
 Each :ref:`applications` and :ref:`samples` in the |NCS| come with their own testing instructions.
-However, the |NCS| provides a common set of features and options for debugging and optimizing the application performance.
+Follow these instructions to make sure that the application runs as expected.
+
+Information about the current state of the application is usually provided through the LEDs or through :term:`Universal Asynchronous Receiver/Transmitter (UART)`, or through both.
+See the user interface section of the application's documentation for description of the LED states or available UART commands.
+
+To see the UART output, connect to the development kit (DK) with a terminal emulator.
+The default serial port settings are the following:
+
+.. list-table:: Default serial port connection settings for Nordic Semiconductor devices
+   :widths: auto
+   :header-rows: 1
+
+   * - Baud rate
+     - Data bits
+     - Stop bits
+     - Parity
+     - :term:`UART Hardware Flow Control (UART HWFC)`
+   * - 115200
+     - 8
+     - 1
+     - No parity
+     - None
+
+Use one of the following methods:
+
+.. tabs::
+
+   .. group-tab:: nRF Connect Serial Terminal
+
+      You can use the `nRF Connect Serial Terminal`_ app, which you can install from `nRF Connect for Desktop`_.
+      The nRF Connect Serial Terminal app establishes serial communication with the device, sends commands through UART, and also displays the UART output.
+
+      The application is available for Windows, Linux, and macOS.
+
+      To connect to DK with Serial Terminal, complete the following steps:
+
+      1. Launch the Serial Terminal app.
+      #. Connect the DK to the PC with a USB cable.
+      #. Power on the DK.
+      #. Click :guilabel:`Select Device` and select the particular kit entry from the drop-down list in the Serial Terminal.
+      #. Select the serial port you want to connect to from the drop-down list.
+      #. Configure the serial port connection parameters.
+         See the `Serial Terminal configuration`_ page for more information.
+      #. Click :guilabel:`Connect to port`.
+
+      nRF Connect Serial Terminal starts the serial communication with the kit.
+
+      For more information about how to connect using the app, see the `Serial Terminal documentation <Connecting using Serial Terminal_>`_.
+
+   .. group-tab:: nRF Connect for Visual Studio Code
+
+      The |nRFVSC| includes an integrated serial port and RTT terminal, which you can use to connect to your board.
+      For detailed instructions, see `How to connect to the terminal`_ on the `nRF Connect for Visual Studio Code`_ documentation site.
+
+      The extension is available for Windows, Linux, and macOS.
+
+   .. group-tab:: PuTTY (Windows only)
+
+      PuTTY is available only for Windows.
```

Comment:
I would like some sort of one-sentence description of PuTTY, to match the previous two tabs.